_overviews/hello-scala/prelude-taste-of-scala.md

```diff
@@ -0,0 +1,532 @@
+```scala
+val d = new Dog("Fido") with TailWagger with Runner
+                        ---------------------------
+```
+
+In that example the traits `TailWagger` and `Runner` are mixed into `Dog` as a new `Dog` instance is created.
+
+
+
+## Collections classes
+
+If you’re coming to Scala from Java and you’re ready to really jump in and learn Scala, here’s a important suggestion about Scala’s collections classes:
+
+>Forget what you know about Java collections classes, and use the Scala collections classes.
+
+You *can* use the Java collections classes in Scala, and many people do so for several months, but when you do that, you’re slowing down your own learning process. A great benefit of the Scala collections classes is that they offer many powerful methods that you’ll want to start using ASAP.
+
+
+### Populating lists
+
+There are times when it’s helpful to create sample lists that are populated with data, and Scala offers many ways to populate lists. Here are just a few:
+
+```scala
+val nums = List.range(0, 10)
+val nums = 1 to 10 by 2 toList
+val letters = ('a' to 'f').toList
+val letters = ('a' to 'f') by 2 toList
+```
+
+
+### Sequence methods
+
+While there are many sequential collections classes you can use — `Array`, `ArrayBuffer`, `Vector`, `List`, and more — let’s look at some examples of what you can do with the `List` class. Given these two lists:
+
+```scala
+val nums = (1 to 10).toList
+val names = List("joel", "ed", "chris", "maurice")
+```
+
+This is the `foreach` method:
+
+```scala
+scala> names.foreach(println)
+joel
+ed
+chris
+maurice
+```
+
+Here’s the `filter` method, followed by `foreach`:
+
+```scala
+scala> nums.filter(_ < 4).foreach(println)
+1
+2
+3
+```
+
+Here are some examples of the `map` method:
+
+```scala
+scala> val doubles = nums.map(_ * 2)
+doubles: List[Int] = List(2, 4, 6, 8, 10, 12, 14, 16, 18, 20)
+
+scala> val capNames = names.map(_.capitalize)
+capNames: List[String] = List(Joel, Ed, Chris, Maurice)
+
+scala> val lessThanFive = nums.map(_ < 5)
+lessThanFive: List[Boolean] = List(true, true, true, true, false, false, false, false, false, false)
+```
+
+Even though we didn’t explain it, you can see how `map` works: It applies an algorithm you supply to every element in the collection, returning a new, transformed value for each element.
+
+If you’re ready to see one of the most powerful collections methods, here’s `reduce`:
+
+```scala
+scala> nums.reduce(_ + _)
+res0: Int = 55
+
+scala> nums.reduce(_ * _)
+res1: Int = 3628800
+```
+
+Again, even though we didn’t explain `reduce`, you can guess that the first example yields the sum of the numbers in `nums`, and the second example returns the product of all those numbers.
+
+There are many (many!) more methods available to Scala collections classes, but hopefully this gives you an idea of their power.
+
+{::comment}
+>There’s so much power in the Scala collections class, over 100 pages are spent discussing them in the *Scala Cookbook*.
+{:/comment}
```

Comment:
Shoot, unfortunately it renders as an HTML comment. I thought it didn’t render in the output at all. That’s a bummer, because while this comment doesn’t matter, I should have deleted it, I do like to keep comments in the text sometimes.

Anyway, skipping all that, it should be deleted.

Reply:
In case you want to add comments, you can use the following syntax:

~~~ markdown
<!--
This is a comment that will not be rendered in the output.
-->
~~~

Reply:
Bummer, when I do a “View Source” it looks like both types of comments are rendered in the HTML output. I’ll try to go without comments, because I don’t like for them to show up in the output.

Reply:
I hope you don’t mind, but I’m going to resolve this conversation, just so it won’t look like an open item.